### ISSUE TITLE
Problem: move transfer can fail silently

### ISSUE THREAD

**Issue by jorikvankemenade** [archivematica/Issues#982]
**Expected behaviour**
When the move transfer job has completed successfully the requested file is moved to the requested location. In all other cases, the job shows a non-zero error code and an error.

**Current behaviour**
The front-end shows that a file has been moved successfully:

![image](https://user-images.githubusercontent.com/7934339/68596531-aef5ae80-049b-11ea-8766-e88b56edb5b9.png)

Inspecting the job we see that a file has been moved from `workFlowDecisions/extractPackagesChoice` to `workFlowDecisions/examineContentsChoice`:

![image](https://user-images.githubusercontent.com/7934339/68596562-bcab3400-049b-11ea-84e0-22ac99d01e18.png)

However. after this point processing stopped. Upon further inspection of the `sharedDirectory` we find:

```shell
watchedDirectories/workFlowDecisions
├── watchedDirectories/workFlowDecisions/compressionAIPDecisions
├── watchedDirectories/workFlowDecisions/createDip
├── watchedDirectories/workFlowDecisions/createTree
├── watchedDirectories/workFlowDecisions/examineContentsChoice
├── watchedDirectories/workFlowDecisions/extractPackagesChoice
│   └── watchedDirectories/workFlowDecisions/extractPackagesChoice/archive_847191-f168aa57-5708-489f-8036-05ca82d7acc5
├── watchedDirectories/workFlowDecisions/metadataReminder
├── watchedDirectories/workFlowDecisions/quarantineTransfer
├── watchedDirectories/workFlowDecisions/selectFormatIDToolIngest
└── watchedDirectories/workFlowDecisions/selectFormatIDToolTransfer
```

The file has not been moved. In the MCP Client logs we find:

```
Nov 11 11:00:55 archivematica-client-dev1.cern.ch python[11464]: archivematicaClient.py: INFO      2019-11-11 10:00:55,164  archivematica.mcp.client:execute_command:206:
Nov 11 11:00:55 archivematica-client-dev1.cern.ch python[11464]: *** RUNNING TASK: movetransfer_v0.0
Nov 11 11:00:55 archivematica-client-dev1.cern.ch python[11464]: Execution failed: [Errno 12] Cannot allocate memory
Nov 11 11:00:55 archivematica-client-dev1.cern.ch python[11464]: archivematicaClient.py: INFO      2019-11-11 10:00:55,197  archivematica.mcp.client:write_task_results_callback:215:
Nov 11 11:00:55 archivematica-client-dev1.cern.ch python[11464]: *** Completed job: #<movetransfer_v0.0; exit=0; code=success uuid=8f741c9c-b749-48bd-a674-5a0cba4b8ab0
Nov 11 11:00:55 archivematica-client-dev1.cern.ch python[11464]: =============== STDOUT ===============
Nov 11 11:00:55 archivematica-client-dev1.cern.ch python[11464]: =============== END STDOUT ===============
Nov 11 11:00:55 archivematica-client-dev1.cern.ch python[11464]: =============== STDERR ===============
Nov 11 11:00:55 archivematica-client-dev1.cern.ch python[11464]: exitCode: -1
Nov 11 11:00:55 archivematica-client-dev1.cern.ch python[11464]: Config Error!
Nov 11 11:00:55 archivematica-client-dev1.cern.ch python[11464]: [Errno 12] Cannot allocate memory
Nov 11 11:00:55 archivematica-client-dev1.cern.ch python[11464]: =============== END STDERR ===============
```

Apparently there has been an error during the execution of `move_transfer` that has not been properly captured.

The script [move_transfer.py](https://github.com/artefactual/archivematica/blob/ef3af9fb58ac700dfca2a350095ef13a0c396e9c/src/MCPClient/lib/clientScripts/move_transfer.py#L56) uses the result of the [`rename`](https://github.com/artefactual/archivematica/blob/ef3af9fb58ac700dfca2a350095ef13a0c396e9c/src/archivematicaCommon/lib/fileOperations.py#L145) to set the result of the job. However, the `rename` function does not return the exit code. Since [`set_status`](https://github.com/artefactual/archivematica/blob/ef3af9fb58ac700dfca2a350095ef13a0c396e9c/src/MCPClient/lib/job.py#L58) defaults to a `success` status this job will **always** succeed.

**Steps to reproduce**
Unfortunately, I have no way of reproducing this reliably.  But given the previous analysis, I think any action that would make the `rename` fail should be enough. 

**Your environment (version of Archivematica, operating system, other relevant details)**
Archivematica 1.10 with Storage Service 0.15 on CentOS 7.

---

**For Artefactual use:**

Before you close this issue, you must check off the following:

- [ ] All pull requests related to this issue are properly linked
- [ ] All pull requests related to this issue have been merged
- [ ] A testing plan for this issue has been implemented and passed (testing plan information should be included in the issue body or comments)
- [ ] Documentation regarding this issue has been written and merged
- [ ] Details about this issue have been added to the release notes (if applicable)

**Comment by replaceafill** [comment]
Just wanted to note that the [`rename` function was recently changed](https://github.com/artefactual/archivematica/commit/2e3cc22de1355f5afc3e1822e8da63de8a870533) in the `qa/1.x` branch and now returns the exit code. I'm not sure about the impact of that change on this issue though.

**Comment by jorikvankemenade** [comment]
@replaceafill thanks for pointing that out. I think that as long as the rest of the workflow is well defined, i.e. a failed move sip results in a failure report this should be sufficient.

**Comment by jorikvankemenade** [comment]
I applied the same change to my local instance and it seems to be detecting non-successful moves. I am not sure if this will always result in a "proper" failure state, haven't tested it very thoroughly,  but it seems alright! 

Will this fix be added to the 1.11 release? If so we might want to keep this issue open for a little bit so that people who run into the same problem can easily find the solution.

**Comment by sevein** [comment]
Fix 2e3cc22de1355f5afc3e1822e8da63de8a870533 was included in v1.12.0.
